salt/containers/config/usr-local-bin-setup-secrets.sh

```diff
@@ -5,6 +5,9 @@ set -e
 #VAULT_ROLE_ID
 #VAULT_SECRET_ID
 
+# store secret id for future logins
+sudo echo "export VAULT_SECRET_ID=$VAULT_SECRET_ID" > /etc/profile.d/vault-secret-id.sh
```

Comment:
Would this break if the secret contained special characters?

Reply:
Not so familiar with `/etc/profile.d`. I understand it won't be read for non-interactive shells?

From the security perspective, I am also wondering whether we would want to make it only available to the user that needs it. And only read it when necessary? Otherwise it seems to be passed to many processes that don't actually need?

Reply:
We [are able](https://github.com/elifesciences/builder-base-formula/blob/5d8102bed07710a7bc3a41d8c57576c789c0c0c7/elife/jenkins-node.sls#L20) to read `/etc/profile`-related things from Jenkins.

We could definitely put the secret somewhere else, it must be readable by the `ubuntu` user which is what Jenkins runs as. That will also remove the `sudo`. The Jenkins step will then do
```
source /somewhere/secret.env && vault.sh read ...
```